jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFComponentProcessor.java

```diff
@@ -253,10 +254,15 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
      * using the facesConfigElement tree it is known when a tag is one of the
      * classes to use for injection.
      */
-    private void processJSFArtifactsForInjection(final DeploymentUnit deploymentUnit, final Set<String> managedBeanClasses) {
+    private void processJSFArtifactsForInjection(final DeploymentUnit deploymentUnit, final Set<String> managedBeanClasses)
+            throws DeploymentUnitProcessingException {
         final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
-        boolean isCDI = support.hasCapability(WELD_CAPABILITY_NAME) &&
-                support.getOptionalCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class).get().isPartOfWeldDeployment(deploymentUnit);
+        if (!support.hasCapability(WELD_CAPABILITY_NAME)) {
```

Comment:
The JSFResourceDefinition should declare an 'org.wildfly.faces' capability that requires WELD_CAPABILITY_NAME. Convert the JSFResourceDefinition c'tor to pass a SimpleResourceDefinition.Parameters object to super(...) and pass that new capability into that object.

Once you have done that there is no longer any possibility that WeldCapability will not be present when a jsf subsystem DUP runs.